dq-game/src/server/controller.py

```diff
@@ -112,20 +113,33 @@ def process_joker(self, message):
         client_name = self.socket.clients[message.origin]["name"]
         self.current_played_jokers[client_name] = message.data
 
-    def ask_question(self, question):
-        # TODO: JOKERS
+    def ask_question(self, question, answer_limit, answer_limit_callback):
         self.current_answers = []
 
         self.socket.send_to_all("ANSWER_QUESTION", "event")
 
         self.is_timeout = False
 
-        while not self.is_timeout:
+        while not self.is_timeout and len(self.current_answers) < answer_limit:
             for message in self.socket.inbuffer:
                 if message.content_type == "data-answer":
                     self.process_answer(message)
                     self.socket.inbuffer.remove(message)
-            time.sleep(0.2)
+                    if len(self.current_answers) == answer_limit:
+                        self.flush_inbuffer()
+                        break
+            time.sleep(0.02)
+
+        if len(self.current_answers) == answer_limit:
+            answer_limit_callback()
```

Comment:
Can you call this function in the line 128, to avoid doing twice the comparisson ?